framework/caching/MemCache.php

```diff
@@ -313,9 +313,10 @@ protected function setValues($data, $duration)
             // @see http://php.net/manual/en/memcache.set.php
             // @see http://php.net/manual/en/memcached.expiration.php
             $expire = $duration > 0 ? $duration + time() : 0;
-            $this->_cache->setMulti($data, $expire);
 
-            return [];
+            // Memcached::setMulti() returns boolean
+            // @see http://php.net/manual/ru/memcached.setmulti.php
+            return $this->_cache->setMulti($data, $expire) ? [] : array_keys($data);
```

Comment:
It isn't necessarily because of `Memcached::setMulti()` always returns `boolean`. See https://github.com/php-memcached-dev/php-memcached/blob/master/php_memcached.c#L1840 .